### PR TITLE
sum_of_geometric_progression

### PR DESCRIPTION
**Pull request by AxSmasher44**
### **Describe your change:**
Implemented algorithm for finding the sum of 'n' terms of a geometric progression

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
Does this work for floating point parameters?  Does it work for negative parameters?

File: maths/sum_of_geometric_progression.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+def sum_of_gp(first_term, common_ratio, num_of_terms):
+    """"
+    Find the sum of n terms in a geometric progression.
+    >>> sum_of_gp(1, 2, 10)
+    1023.0
+    >>> sum_of_gp(1, 10, 5)
+    11111.0
+    """
+    if common_ratio == 1:
+        sum = num_of_terms * first_term
```
Comment (cclauss) [inline maths/sum_of_geometric_progression.py:10]: [sum()](https://docs.python.org/3/library/functions.html#sum) is a Python builtin so it should not be used as a variable name.

```suggestion
        return num_of_terms * first_term
```

File: maths/sum_of_geometric_progression.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    1023.0
+    >>> sum_of_gp(1, 10, 5)
+    11111.0
+    """
+    if common_ratio == 1:
+        sum = num_of_terms * first_term
+        return sum
+    # formula for sum if common ratio is 1
+
+    sum = (first_term / (1 - common_ratio)) * (1 - common_ratio ** num_of_terms)
```
Comment (cclauss) [inline maths/sum_of_geometric_progression.py:14]:
```suggestion
    return (first_term / (1 - common_ratio)) * (1 - common_ratio ** num_of_terms)

```

**Comment by AxSmasher44** [comment]
> Does this work for floating point parameters? Does it work for negative parameters?

Yes, it works for floating point parameters  and negative parameters.

File: maths/sum_of_geometric_progression.py [hunk 3]
```diff
@@ -7,7 +7,7 @@ def sum_of_gp(first_term, common_ratio, num_of_terms):
     11111.0
     """
     if common_ratio == 1:
-        sum = num_of_terms * first_term
+        return num_of_terms * first_term
         return sum
```
Comment (cclauss) [inline maths/sum_of_geometric_progression.py]:
```suggestion
```

**Comment by cclauss** [comment]
Do these tests with zeros and negative numbers make sense or is this garbage data that should be detected and rejected with a `raise ValueError()`?

**Comment by cclauss** [comment]
./maths/sum_of_geometric_progression.py:1:89: E501 line too long (97 > 88 characters)

**Comment by divyanshugoyal103** [comment]
float case will fail I think

**Pull request merged**